test/lit/metadce/functions_with_names_need_escape.wat

```diff
@@ -0,0 +1,34 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+;; RUN: wasm-metadce %s --graph-file %s.json -S -o - | filecheck %s --check-prefix=TXT
+;; RUN: wasm-as %s -o %t.wasm --source-map %t.map
+;; RUN: wasm-metadce %t.wasm --input-source-map %t.map --graph-file %s.json -o %t.out.wasm --output-source-map %t.out.map
+;; RUN: wasm-dis %t.out.wasm --source-map %t.out.map -o - | filecheck %s --check-prefix=BIN
+
+;; Test that sourcemap information is preserved
+
+(module
+  ;;@ a:1:2
+  ;; TXT:      (type $0 (func))
+
+  ;; TXT:      (export "f" (func $f))
+
+  ;; TXT:      (func $f
+  ;; TXT-NEXT:  ;;@ a:7:8:someSymbol
+  ;; TXT-NEXT:  (nop)
+  ;; TXT-NEXT:  ;;@ a:9:10
+  ;; TXT-NEXT: )
+  (func $f (export "f")
+     ;;@ a:7:8:someSymbol
```

Comment:
The source-map stuff (`;;@` comments) can be removed, as we don't need to test that here (IIUC).